src/main/java/seedu/address/model/commons/NameContainsKeywordsPredicate.java

```diff
@@ -4,22 +4,30 @@
 import java.util.function.Predicate;
 
 import seedu.address.commons.util.StringUtil;
-import seedu.address.model.activity.Activity;
+import seedu.address.model.travelplan.TravelPlan;
 
 /**
- * Tests that a {@code Activity}'s {@code Name} matches any of the keywords given.
+ * Tests that a {@code TravelPlanObject}'s or {@code TravelPlan}'s {@code Name} matches any of the keywords given.
  */
-public class NameContainsKeywordsPredicate implements Predicate<Activity> {
+public class NameContainsKeywordsPredicate implements Predicate<Object> {
     private final List<String> keywords;
 
     public NameContainsKeywordsPredicate(List<String> keywords) {
         this.keywords = keywords;
     }
 
     @Override
-    public boolean test(Activity activity) {
-        return keywords.stream()
-                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(activity.getName().activityName, keyword));
+    public boolean test(Object obj) {
+        if (obj instanceof TravelPlan) {
+            return keywords.stream()
+                    .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(((TravelPlan) obj).getName().name, keyword));
+        } else if (obj instanceof TravelPlanObject) {
+            return keywords.stream()
+                    .anyMatch(keyword ->
+                            StringUtil.containsWordIgnoreCase(((TravelPlanObject) obj).getName().name, keyword));
+        } else {
+            return false;
+        }
```

Comment:
As for this, 2 separate (overloaded) test methods:
`public boolean test(TravelPlan travelPlan)` and `public boolean test(TravelPlanObject tpObject)`